pkg/resources/function_commons.go

```diff
@@ -0,0 +1,344 @@
+		Required:         true,
+		ForceNew:         true,
+		DiffSuppressFunc: suppressIdentifierQuoting,
+		Description:      blocklistedCharactersFieldDescription("The database in which to create the function."),
+	},
+	"schema": {
+		Type:             schema.TypeString,
+		Required:         true,
+		ForceNew:         true,
+		DiffSuppressFunc: suppressIdentifierQuoting,
+		Description:      blocklistedCharactersFieldDescription("The schema in which to create the function."),
+	},
+	"name": {
+		Type:             schema.TypeString,
+		Required:         true,
+		Description:      blocklistedCharactersFieldDescription("The name of the function; the identifier does not need to be unique for the schema in which the function is created because UDFs are identified and resolved by the combination of the name and argument types. Check the [docs](https://docs.snowflake.com/en/sql-reference/sql/create-function#all-languages)."),
+		DiffSuppressFunc: suppressIdentifierQuoting,
+	},
+	"is_secure": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		Default:          BooleanDefault,
+		ValidateDiagFunc: validateBooleanString,
+		DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInShow("is_secure"),
+		Description:      booleanStringFieldDescription("Specifies that the function is secure. By design, the Snowflake's `SHOW FUNCTIONS` command does not provide information about secure views (consult [function docs](https://docs.snowflake.com/en/sql-reference/sql/create-function#id1) and [Protecting Sensitive Information with Secure UDFs and Stored Procedures](https://docs.snowflake.com/en/developer-guide/secure-udf-procedure)) which is essential to manage/import function with Terraform. Use the role owning the function while managing secure functions."),
+	},
+	"is_aggregate": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		Default:          BooleanDefault,
+		ValidateDiagFunc: validateBooleanString,
+		DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInShow("is_aggregate"),
+		Description:      booleanStringFieldDescription("Specifies that the function is an aggregate function. For more information about user-defined aggregate functions, see [Python user-defined aggregate functions](https://docs.snowflake.com/en/developer-guide/udf/python/udf-python-aggregate-functions)."),
+	},
+	"arguments": {
+		Type: schema.TypeList,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"arg_name": {
+					Type:     schema.TypeString,
+					Required: true,
+					// TODO [SNOW-1348103]: adjust diff suppression accordingly.
+					Description: "The argument name.",
+				},
+				// TODO [SNOW-1348103]: after testing weird names add limitations to the docs and add validation here
+				"arg_data_type": {
+					Type:             schema.TypeString,
+					Required:         true,
+					ValidateDiagFunc: IsDataTypeValid,
+					DiffSuppressFunc: DiffSuppressDataTypes,
+					Description:      "The argument type.",
```

Comment:
Nit: add a link to Snowflake docs.

Reply:
There is a link in description for the wrapping "arguments".

Reply:
I meant the link to https://docs.snowflake.com/en/sql-reference/intro-summary-data-types.